src/app/services/bis.ts

```diff
@@ -847,6 +850,52 @@ export const api = createApi({
           : []
         ).concat([{ type: 'Participant' as const, id: 'PARTICIPANT_LIST' }]),
     }),
+    readAttendanceListPages: build.query<
+      PaginatedList<AttendanceListPage>,
+      ListArguments & { eventId: number }
+    >({
+      query: queryArg => ({
+        url: `frontend/events/${queryArg.eventId}/record/attendance_list_pages/`,
+        params: {
+          page: queryArg.page,
+          page_size: queryArg.pageSize,
+          search: queryArg.search,
+        },
+      }),
+    }),
+    createAttendanceListPage: build.mutation<
+      AttendanceListPage,
+      { eventId: number; eventAttendanceListPage: AttendanceListPagePayload }
+    >({
+      query: queryArg => ({
+        url: `frontend/events/${queryArg.eventId}/record/attendance_list_pages/`,
+        method: 'POST',
+        body: queryArg.eventAttendanceListPage,
+      }),
+    }),
+    updateAttendanceListPage: build.mutation<
+      AttendanceListPage,
+      {
+        eventId: number
+        id: number
+        patchedAttendanceListPage: Partial<AttendanceListPagePayload>
+      }
+    >({
+      query: queryArg => ({
+        url: `frontend/events/${queryArg.eventId}/record/attendance_list_pages/${queryArg.id}/`,
+        method: 'PATCH',
+        body: queryArg.patchedAttendanceListPage,
+      }),
+    }),
+    deleteAttendanceListPage: build.mutation<
+      void,
+      { eventId: number; id: number }
+    >({
+      query: queryArg => ({
+        url: `frontend/events/${queryArg.eventId}/record/attendance_list_pages/${queryArg.id}/`,
+        method: 'DELETE',
+      }),
+    }),
```

Comment:
Endpoints look good! The only issue is, when attendance list page is created/updated/deleted, the change isn't shown. It's due to cache capabilities of rtk-query. So, this could be fixed with adding [`providesTags` and `invalidatesTags`](https://redux-toolkit.js.org/rtk-query/usage/automated-refetching) to new endpoint definitions, analogously to the event photo endpoints below.

Reply:
Added the tags in 2cf5f5bc2a8f0e52e1934737d36bf97658698f0b